src/components/PostDetails.tsx

```diff
@@ -1,106 +1,94 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { CommentsList } from './CommentsList';
+
+import { deleteComment, getComments } from '../api/comments';
+
+interface Props {
+  selectedPost: Post;
+}
+
+export const PostDetails: FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [error, setError] = useState('');
+  const [isShownForm, setIsShownForm] = useState(false);
+
+  const handleDeleteComment = (id: Comment['id']) => {
+    deleteComment(id).then(() =>
+      setComments(currentComments =>
+        currentComments.filter(comment => comment.id !== id),
+      ),
+    );
```

Comment:
The `handleDeleteComment` function lacks error handling. Consider adding a `catch` block to handle potential errors during the deletion process and inform the user appropriately.